### PR TITLE
#448 - support ordinal-N prefix in brokerProperties for pod specific …

### PR DESCRIPTION
**Pull request by gtully**
…broker behaviours

### CODE REVIEW

**Comment by brusdev** [comment]
@gtully nice improvement!!!

WDYT about the following alternative prefixes?
- broker-N
- $broker-N
- $N

I would like `broker-N`, i.e. `broker-0.connectionRouters.autoShard.localTargetFilter=NULL|0|-0`

**Comment by gtully** [comment]
I like it, the ordinal-N is a bit clunky - however it is indeed what it is tied to, the stateful set ordinal, and it is unlikely to ever be a broker configuration attribute.

But broker-N. is indeed better. I don't want the use $ as that is the default for env var expansion.
thanks for the fast feedback :-) I will make it broker-N

**Comment by gtully** [comment]
broker-<stateful set ordinal> to be more exact

File: controllers/activemqartemis_controller_test.go [hunk 1]
```diff
@@ -6309,6 +6307,151 @@ var _ = Describe("artemis controller", func() {
 		}, timeout, interval).Should(BeTrue())
 	})
 
+	It("ordinal broker properties", func() {
+		ctx := context.Background()
+		crd := generateArtemisSpec(defaultNamespace)
+		crd.Spec.DeploymentPlan.Size = 2
+		crd.Spec.BrokerProperties = []string{
+			"name=BROKER_ORDINAL_0",
```
Comment (gaohoward) [inline controllers/activemqartemis_controller_test.go:6315]: Is this 'name' only overrides pod-0 or all the broker-n (except broker-1)? I think some doc need here to explain the behavior. It looks like if you scale up to 3 the 3rd doesn't use this name. (I saw some log like File /amq/extra/secrets/artemis-broker-props/broker-2 does not exist), maybe it's a bug.

Comment (gtully) [reply controllers/activemqartemis_controller_test.go:6315]: properties without a prefix end up in <mount point>/broker.properties in the normal way. Properties with a prefix end up in <mount-point>/broker-N/broker-N.broker.properties and each broker looks in mount-point and mount-point/broker-${STATEFUL_SET_ORDINAL} - such that it only picks up the prefixed properties that match.

let me check with a test for scale up, I think the primary use case is static pods, however any pod that does not have extra prefixed props, will get the normal config.

Comment (gtully) [reply controllers/activemqartemis_controller_test.go:6315]: and I will add some doc, in the help/operator.md - there is currently no brokerProperties section, I will fix that. thanks

Comment (gtully) [reply controllers/activemqartemis_controller_test.go:6473]: it is better now

**Review by brusdev — APPROVED** [review]

**Review by gaohoward — APPROVED** [review]

**Pull request merged**